sdk/core/azure-core/inc/credentials.hpp

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#ifndef _az_CORE_IDENTITY_CREDENTIALS_HPP
+#define _az_CORE_IDENTITY_CREDENTIALS_HPP
+
+#include <chrono>
+#include <string>
+
+#include "_internal/prefix.hpp"
+
+namespace azure
+{
+
+namespace identity
+{
+
+class credential;
+
+namespace _internal
+{
+void set_scopes(credential& credential, std::string const& scopes);
+} // namespace _internal
+
+class credential
+{
+  friend void _internal::set_scopes(credential&, std::string const&);
+
+public:
+  virtual ~credential() noexcept {};
+
+protected:
+  credential() = default;
+
+private:
+  credential(credential const&) = delete;
```

Comment:
I see you're marking the copy constructor and assignment operator deleted for all classes. What's the point of doing so? Is this part of some guidelines?

Reply:
Because we (at least currently) can't guarantee it will work correctly, so we disable copy and assignment.